### ISSUE TITLE
Build endpoint abruptly closes connection when tag param is invalid

### ISSUE THREAD

**Issue by shin-** [docker/docker#21069]
Originally reported in https://github.com/docker/docker-py/issues/978

Expected result: HTTP status 400 "Bad request", (ideally) with explicit error message about invalid parameter.

Observed result: Connection aborted - Connection reset by peer.

**Comment by yongtang** [comment]
I tried myself on 1.10.3 and current master, it seems that this has already been addressed:
```
$ python dockerbuilder.py 
Traceback (most recent call last):
  File "dockerbuilder.py", line 10, in <module>
    rm=True)]
  File "/usr/local/lib/python2.7/dist-packages/docker/client.py", line 230, in _stream_helper
    yield self._result(response)
  File "/usr/local/lib/python2.7/dist-packages/docker/client.py", line 150, in _result
    self._raise_for_status(response)
  File "/usr/local/lib/python2.7/dist-packages/docker/client.py", line 146, in _raise_for_status
    raise errors.APIError(e, response, explanation=explanation)
docker.errors.APIError: 500 Server Error: Internal Server Error ("Error parsing reference: "ESSAI" is not a valid repository/tag")
```

Notice the last line: `Error parsing reference: "ESSAI" is not a valid repository/tag`

**Comment by thaJeztah** [comment]
Thanks @yongtang 

@shin- can you confirm it's fixed, then we can close this

**Comment by shin-** [comment]
Yes, looks like this is fixed in 1.11.0rc1 - Thank you!
